lib/segment/src/types.rs

```diff
@@ -1647,6 +1673,101 @@ mod tests {
         eprintln!("de_record = {de_record:#?}");
     }
 
+    #[test]
+    fn test_geo_radius_check_point() {
+        let radius = GeoRadius {
+            center: GeoPoint { lon: 0.0, lat: 0.0 },
+            radius: 80000.0,
+        };
+
+        let inside_result = radius.check_point(0.5, 0.5);
+        assert!(inside_result);
+
+        let outside_result = radius.check_point(1.5, 1.5);
+        assert!(!outside_result);
+    }
+
+    #[test]
+    fn test_geo_boundingbox_check_point() {
```

Comment:
Maybe we could give it a more precise name as it assess the edge case calculation over the prime meridian.

Reply:
discarding comment